src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -93,13 +92,16 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                     {
                         if (docOptions.GetOption(CodeCleanupOptions.AreCodeCleanupRulesConfigured))
                         {
-                            // Code cleanup
-                            var oldDoc = document;
-                            var codeCleanupChanges = GetCodeCleanupAndFormatChangesAsync(document, codeCleanupService, cancellationToken).WaitAndGetResult(cancellationToken);
-
-                            if (codeCleanupChanges != null && codeCleanupChanges.Count() > 0)
+                            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.This_can_take_several_seconds_when_there_are_many_changes))
                             {
-                                ApplyChanges(oldDoc, codeCleanupChanges.ToList(), selectionOpt: null, cancellationToken);
+                                // Code cleanup
+                                var oldDoc = document;
+                                var codeCleanupChanges = GetCodeCleanupAndFormatChangesAsync(document, codeCleanupService, cancellationToken).WaitAndGetResult(cancellationToken);
+
+                                if (codeCleanupChanges != null && codeCleanupChanges.Count() > 0)
```

Comment:
this is potentially o(n) since this is a lazy list.  please realize the list first (i.e. .ToList()) then check.